status-panels-lib/README.md

```diff
@@ -0,0 +1,161 @@
+# Status panels lib
+
+This status panels lib can be used to add a status panel row for integration dashboards using [grafonnet](https://github.com/grafana/grafonnet).
+
+The status panels provide the following information for an integration.
+
+- Metrics status
+  - Shows if the agent is sending metrics for this integration
+  - Shows the latest timestamp the metrics were received for this integration
+- Logs status
+  - Shows if the agent is sending logs for this integration
+  - Shows the latest timestamp the logs were received for this integration
+- Integration version
+  - Shows the current installed version of the integration
+
+## Import
+
+```sh
+jb init
+jb install https://github.com/grafana/jsonnet-libs/status-panels-lib
+```
+
+# Usage
+
+## Metrics
+
+```
+(statusPanels.new(
+  'Integration Status',
+  type='metrics',
+  statusPanelsQueryMetrics='up{job=~"$job"}',
+  datasourceNameMetrics='$prometheus_datasource',
+  showIntegrationVersion=true,
+  integrationVersion='x.x.x',
+  panelsHeight=2,
+  panelsWidth=8,
+  rowPositionY=10,
+)).panels.statusPanelsWithRow
+```
+
+## Logs
+
+```
+(statusPanels.new(
+  'Integration Status',
+  type='logs',
+  statusPanelsQueryLogs='count_over_time{job=~"$job"}',
+  datasourceNameLogs='$loki_datasource',
+  showIntegrationVersion=true,
+  integrationVersion='x.x.x',
+  panelsHeight=2,
+  panelsWidth=8,
+  rowPositionY=10,
+)).panels.statusPanelsWithRow
+```
+
+## Only panels without row
+
+```
+(statusPanels.new(
+  'Integration Status',
+  type='logs',
+  statusPanelsQueryLogs='count_over_time{job=~"$job"}',
+  datasourceNameLogs='$loki_datasource',
+  showIntegrationVersion=true,
+  integrationVersion='x.x.x',
+  panelsHeight=2,
+  panelsWidth=8,
+  rowPositionY=10,
+)).panels.statusPanels
+```
+
+## Options
+
+### title
+
+Title of the status panels row
+
+### type
+
+`metrics`, `logs` or `both` (Default `metrics`)
+When using `metrics` provide `statusPanelsQueryMetrics` and `datasourceNameMetrics`
+When using `logs` provide `statusPanelsQueryLogs` and `datasourceNameLogs`
+When using `both` provide both metrics & logs query & data source names
+
+### statusPanelsQueryMetrics
+
+Query for checking the status of the integration (Should be the most commonly available metric which is almost always available like up{job="integrations/my-integration"})
+
+### statusPanelsQueryLogs
+
+Query for checking the status of the integration (Should be the most commonly available metric which is almost always available like count_over_time{job="integrations/my-integration"})
+
+### datasourceNameMetrics
+
+Name of the data source to be used with the query targets for metrics
+
+### datasourceNameLogs
+
+Name of the data source to be used with the query targets for logs
+
+### showIntegrationVersion
+
+Whether to show or hide the integration version panel (Default `true`)
+
+### integrationVersion
+
+Pass integration version in "x.x.x" format (Applicable if `showIntegrationVersion` is set to true)
```

Comment:
Not really feedback for changes in this PR, but.. This provides a vector for human error. One might remember to update the metadata, but not make the change here.

This means we now have the Changelog, the metadata, and a couple lines in the mixin file to change for the same value.

We might consider fetching this from one place and distributing it, but that's a larger task.

Reply:
Your are right, we should always gather this info from metadata.json. The status queries will come from there, so this will need to be imported either way.

Reply:
Turns out that Shahzeb is way ahead of us on this. He's already doing that in the Docker PR, by referencing it in the .